nameresolution/kubernetes/kubernetes.go

```diff
@@ -45,10 +45,13 @@ func (k *resolver) Init(metadata nameresolution.Metadata) error {
 	if err != nil {
 		return err
 	}
-	if config, ok := configInterface.(map[string]string); ok {
-		clusterDomain := config[ClusterDomainKey]
-		if clusterDomain != "" {
-			k.clusterDomain = clusterDomain
+	if config, ok := configInterface.(map[string]interface{}); ok {
+		clusterDomainPtr := config[ClusterDomainKey]
```

Comment:
`Ptr` is a bad name for this. It's not a pointer but an interface value that has not been cast yet.

Can you please change the variable name?